src/occupation.jl

```diff
@@ -137,34 +137,39 @@ function compute_fermi_level(basis::PlaneWaveBasis{T}, eigenvalues, ::FermiZeroT
                              temperature, smearing, tol_n_elec) where {T}
     filled_occ  = filled_occupation(basis.model)
     n_electrons = basis.model.n_electrons
-    n_spin = basis.model.n_spin_components
     @assert iszero(temperature)
 
     # Sanity check that we can indeed fill the appropriate number of states
-    if n_electrons % (n_spin * filled_occ) != 0
+    if n_electrons % (filled_occ) != 0
         error("$n_electrons electrons cannot be attained by filling states with " *
               "occupation $filled_occ. Typically this indicates that you need to put " *
               "a temperature or switch to a calculation with collinear spin polarization.")
     end
-    n_fill = div(n_electrons, n_spin * filled_occ, RoundUp)
-
-    # For zero temperature, two cases arise: either there are as many bands
-    # as electrons, in which case we set εF to the highest energy level
-    # reached, or there are unoccupied conduction bands and we take
-    # εF as the midpoint between valence and conduction bands.
-    if n_fill == length(eigenvalues[1])
-        εF = maximum(maximum, eigenvalues) + 1
-        εF = mpi_max(εF, basis.comm_kpts)
+
+    all_eigenvalues = sort(vcat(eigenvalues...))
```

Comment:
Careful that this will not work with MPI (hopefully the tests should catch this)

Reply:
You need to wrap and call MPI.Allgatherv here, so that each process does a bisection on the full array. (You cannot do a Gatherv because excess_n_electrons calls mpi_sum and will hang if not all processes participate.)